client/api/request.js

```diff
@@ -0,0 +1,43 @@
+import parseJson from 'lib/utils/parse-json';
```

Comment:
The idea that I have floating in my mind about code reuse is something like this: The code for Calypso and for our plugin will be exactly the same, except for some files. So, for example, you will have:
* client/api/request.calypso.js
* client/api/request.js

And `Webpack` trickery will pick `request.calypso.js` instead of `request.js` if the environment is Calypso. Probably it will be the other way around (I would prefer to mess with our Webpack config instead of Calypso's), but you get the general idea.

This file (`request.js`) has the WP-Admin specific logic to make REST requests. Calypso won't use `nonce`, will use another way to construct the request themselves, and won't have the `window.persistState` logic. But the other API-related files (`api/index.js`, `api/url.js`) will be common for both the systems.

The refactor will continue in that direction: pushing the code that will differ between Calypso and WP-Admin to small, isolated files.